spec/views/hq/users/index_spec.rb

```diff
@@ -2,51 +2,62 @@
 
 describe "hq/users/index.html.erb", type: :view do
 
-  it 'should have a New User link' do
-    assign(:users, [])
-    render
-    expect(rendered).to have_link('New User', new_hq_user_path)
+  context 'class action-items' do
+    it 'should have a New User link' do
+      assign(:users, User.none.paginate(:page => 1))
+      render
+
+      expect(rendered).to have_link('New User', new_hq_user_path)
+    end
   end
 
   describe 'should display users correctly' do
-
     context 'no users' do
-      it 'displays just the header row' do
-        @users = []
+      it 'hides the table' do
+        @users = User.none.paginate(:page => 1)
         render
-        expect(rendered).to have_selector('table tr', count: 1)
+
+        expect(rendered).to match /No Users Found/
+        expect(rendered).to have_selector('table#users-index', count: 0)
       end
     end
 
     context '1 user' do
       it 'displays a single user' do
-        @users = [ double('User', user_name: 'Frederick Bloggs', 
-          email: 'fred.bloggs@example.com', active_sponsors: (1..5).to_a)]
+        @users = [ FactoryGirl.build_stubbed(:user, user_name: 'Frederick Bloggs',
+          email: 'fred.bloggs@example.com') ].paginate
         render
+
         expect(rendered).to have_selector('table tr', count: 2)
         expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
         expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
       end
     end
 
     context '>1 user' do
+      before :each do
+        @users= [ FactoryGirl.build_stubbed(:user),
+                  FactoryGirl.build_stubbed(:user, user_name: 'Joseph Soap',
+                  email: 'joe.soap@example.com'),
+                  FactoryGirl.build_stubbed(:user, user_name: 'Tom Jones',
+                  email: 'tom.jones@example.com') ].paginate(:page => 2, :per_page => 2)
+      end
+
       it 'displays multiple users' do
-        @users = [ double('User', user_name: 'Frederick Bloggs', 
-          email: 'fred.bloggs@example.com', active_sponsors: (1..5).to_a), 
-                   double('User', user_name: 'Joseph Soap', 
-          email: 'joe.soap@example.com', active_sponsors: (1..7).to_a) ]
         render
-        expect(rendered).to have_selector('table tr', count: 3)
-        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
-        expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
-        expect(rendered).to have_link('Joseph Soap', href: hq_user_path(@users[1]))
-        expect(rendered).to match 'joe.soap@example.com'
-        expect(rendered).to match '7'
+
+        expect(rendered).to have_selector('table tr', count: 2)
```

Comment:
because first tr is the table head.
We can use:
`expect(rendered).to have_selector('table tbody tr', count: 1)`